internal/runners/manifest/manifest.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type primeable interface {
+	Output() output.Outputer
+	Project() *project.Project
+	Auth() *authentication.Auth
+	Analytics() analytics.Dispatcher
+	SvcModel() *model.SvcModel
+	Config() *config.Instance
+}
+
+type Manifest struct {
+	out       output.Outputer
+	project   *project.Project
+	auth      *authentication.Auth
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+	cfg       *config.Instance
+}
+
+func NewManifest(prime primeable) *Manifest {
+	return &Manifest{
+		out:       prime.Output(),
+		project:   prime.Project(),
+		auth:      prime.Auth(),
+		analytics: prime.Analytics(),
+		svcModel:  prime.SvcModel(),
+		cfg:       prime.Config(),
+	}
+}
+
+func (m *Manifest) Run() (rerr error) {
+	if m.project == nil {
+		return rationalize.ErrNoProject
+	}
+
+	m.out.Notice(locale.Tl("manifest_operating_on_project", "Operating on project: [ACTIONABLE]{{.V0}}[/RESET], located at [ACTIONABLE]{{.V1}}[/RESET]\n", m.project.Namespace().String(), m.project.Dir()))
+
+	commitID, err := localcommit.Get(m.project.Dir())
+	if err != nil {
+		return errs.Wrap(err, "Could not get commit ID")
+	}
+
+	bp := model.NewBuildPlannerModel(m.auth)
+	expr, _, err := bp.GetBuildExpressionAndTime(commitID.String())
+	if err != nil {
+		return errs.Wrap(err, "Could not get remote build expr and time")
+	}
+
+	exprReqs, err := expr.Requirements()
+	if err != nil {
+		return errs.Wrap(err, "Could not get requirements")
+	}
+
+	artifacts, err := m.fetchArtifacts()
+	if err != nil {
+		return errs.Wrap(err, "Could not fetch artifacts")
+	}
+
+	var vulns vulns
+	if m.auth.Authenticated() {
+		vulns, err = m.fetchVulns(exprReqs)
+		if err != nil {
+			return errs.Wrap(err, "Could not fetch vulnerabilities")
+		}
+	}
+
+	reqs, err := newRequirementsOutput(exprReqs, artifacts, vulns, m.auth)
+	if err != nil {
+		return errs.Wrap(err, "Could not get requirements output")
+	}
+
+	m.out.Print(reqs)
+
+	return nil
+}
+
+func (m *Manifest) fetchArtifacts() ([]*artifact.Artifact, error) {
+	if m.project == nil || strings.EqualFold(os.Getenv(constants.DisableRuntime), "true") {
+		return nil, nil
+	}
+
+	target := target.NewProjectTarget(m.project, nil, target.TriggerPackage)
+	rt, err := runtime.New(target, m.analytics, m.svcModel, m.auth, m.cfg, m.out)
+	if err != nil {
+		return nil, locale.WrapError(err, "err_packages_update_runtime_init", "Could not initialize runtime.")
+	}
+
+	// This shouldn't happen, but it's good to warn the user if it does
+	if rt.NeedsUpdate() {
+		m.out.Notice(locale.T("manifest_runtime_needs_update"))
+	}
+
+	return rt.ResolvedArtifacts()
+}
+
+type vulns map[string]*model.VulnerabilityIngredient
+
+func (v vulns) getVulns(name, namespace string) (*model.VulnerabilityIngredient, bool) {
```

Comment:
Please use fully written words for symbols that aren't scoped to a small set of logic.

Also the name suggests multiple are returned, but it only deals with one.

This also goes for addVulns and fetchVulns.